### ISSUE TITLE
Add new observation type: time bound

### ISSUE THREAD

**Issue by jamessynge** [panoptes/POCS#496]
I'd like to be able to specify that a specific target is to be observed for a certain period of time (e.g. starting an hour before the start of a known transit, continuing until an hour hour after it finishes the transit).

**Comment by wtgee** [comment]
Yes, we've wanted a Time Constraint for a while. Note that much of the work has been done here: https://astroplan.readthedocs.io/en/latest/tutorials/periodic.html

**Comment by wtgee** [comment]
See https://astroplan.readthedocs.io/en/latest/tutorials/periodic.html
